modules/fluid_properties/src/userobjects/SalineFluidProperties.C

```diff
@@ -0,0 +1,146 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+//
+#include "SalineFluidProperties.h"
+
+registerMooseObject("FluidPropertiesApp", SalineFluidProperties);
```

Comment:
Check out https://github.com/idaholab/moose/pull/23019/files for the approach taken for the optional Thermochimica coupling. There we also always register objects, but we indicate a missing Thermochimica library in the class description, too.

Reply:
Are you talking about the calls to `checkLibraryAvailability` and associated error handling?

Reply:
there's also a addClassDescription.

@dschwen from our discussion there, should we copy-paste this ThermochimicaUtils as a SalineUtils?
I think @rks171 achieved most of the same results with just preprocessor directives directly in the object. I think making a util made more sense for thermochimica because there was more than one object

Reply:
I reproduced the other behavior that the utils provide in the object. The checkAvailability was already done

Reply:
resolving